condition_body_array_match.go

```diff
@@ -0,0 +1,88 @@
+package ladon
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/d3sw/ladon/compiler"
+)
+
+const (
+	Matchall = "all"
+	Matchany = "any"
+)
+
+// BodyArrayMatchCondition is a condition which is fulfilled if the value at the
+// path in the body is an array and all/any of its elements match the regex
+// pattern specified in BodyArrayMatchCondition
+type BodyArrayMatchCondition struct {
+	Mode    string `json:"mode"`
+	Path    string `json:"path"`
+	Matches string `json:"matches"`
+}
+
+// Fulfills returns true if the value at the path in the body is an array and
+// all/any of its elements match the regex pattern specified in BodyMatchCondition
+func (c *BodyArrayMatchCondition) Fulfills(_ interface{}, r *Request) bool {
+	req, ok := r.Context[KeyRawRequest].(*http.Request)
+	if !ok {
+		return false
+	}
+	contentType := strings.ToLower(req.Header.Get("Content-type"))
+	switch contentType {
+	case "application/json":
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			return false
+		}
+		req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
Is the wrapping of the buffer needed here?  You should be able to do without

Reply:
Request.Body is of type io.ReadCloser and ioutil.NopCloser is to provides a no-op Close method to bytes.Buffer.